src/components/TestToolMenu.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import styles from '../styles/styles';
+import Switch from './Switch';
+import Text from './Text';
+import * as User from '../libs/actions/User';
+import * as Network from '../libs/actions/Network';
+import * as Session from '../libs/actions/Session';
+import ONYXKEYS from '../ONYXKEYS';
+import Button from './Button';
+import * as NetworkStore from '../libs/Network/NetworkStore';
+import TestToolRow from './TestToolRow';
+
+const propTypes = {
+    /** User object in Onyx */
+    user: PropTypes.shape({
+        /** Whether we should use the staging version of the secure API server */
+        shouldUseSecureStaging: PropTypes.bool,
+    }),
+
+    /** Network object in Onyx */
+    network: PropTypes.shape({
+        /** Whether we should fail all network requests */
+        shouldFailAllRequests: PropTypes.bool,
+    }),
+};
+
+const defaultProps = {
+    user: {
+        shouldUseSecureStaging: false,
+    },
+    network: {
+        shouldFailAllRequests: false,
+    },
+};
+
+const TestToolMenu = props => (
+    <>
+        <Text style={[styles.formLabel]} numberOfLines={1}>
+            Test Preferences
+        </Text>
+
+        {/* Option to switch from using the staging secure endpoint or the production secure endpoint.
+        This enables QA and internal testers to take advantage of sandbox environments for 3rd party services like Plaid and Onfido. */}
+        <TestToolRow title="Use Secure Staging Server">
+            <Switch
+                isOn={props.user.shouldUseSecureStaging}
```

Comment:
When I tested locally, I got this console warning:

<img width="653" alt="Screen Shot 2022-04-22 at 16 34 55" src="https://user-images.githubusercontent.com/6829422/164802508-2533a650-5acd-4f42-a3d0-d4be03e591ed.png">

I think we can fix that by adding the default values to `Onyx.init`, no?

https://github.com/Expensify/App/blob/5b8b75cca48740db41a6af50452c9312e3e80510/src/setup/index.js#L26-L36

Reply:
Oh hmm I guess if there is a `user` object then it won't use the defaults. Sorry @Justicea83 I am going to put the short circuit back in here. @marcochavezf not sure if this is something for `initialKeyStates`- the code was fine before so I'm going to revert that last change.

Reply:
Sure that's fine, we look good now!